src/CycleGraph.jl

```diff
@@ -76,95 +76,84 @@ function LG.has_edge(g::CycleGraph{T}, u, v) where {T}
-
-    e = LG.Edge{T}(one(T), T(2))
-    nvg == T(2) && return e, T(2)
-
-    return e, T(1) 
+    return (ne(g), )
 end
 
-function LG.iterate(iter::LG.SimpleGraphs.SimpleEdgeIter{G}, state) where {G <: CycleGraph}
+# TODO propagate inbounds
+function Base.getindex(edgevec::SimpleEdgeVector{V, G}, i::Int) where {V, G <: CycleGraph}
 
-    g = iter.g
-    nvg = nv(g)
-    T = eltype(g) 
+    i ∈ Base.OneTo(length(edgevec)) || throw(BoundsError(edgevec, i))
 
-   state == nvg && return nothing
-    src = state
-    dst = ifelse(state == one(T), nvg, state + one(T))
-    e = LG.Edge{T}(src, dst)
-        
-    return e, (state + one(T))
+    g = edgevec.graph
+    T = eltype(g)
+    nvg::T = nv(g)
 
+    i == 1 && return LG.Edge(one(T), T(2))
+    i == 2 && return LG.Edge(one(T), nvg)
+    return LG.Edge(T(i - 1), T(i))
 end
 
+Base.IndexStyle(::Type{<:SimpleEdgeVector{V, G}}) where {V, G <: CycleGraph} = IndexLinear()
+
 
 # =======================================================
 #        neighbors
 # =======================================================
 
 # TODO maybe we want an inbounds check
-LG.outneighbors(g::CycleGraph, v::Integer) = OutNeighborsIter(g, eltype(g)(v))
+LG.outneighbors(g::CycleGraph, v::Integer) = OutNeighborVector(g, eltype(g)(v))
 
 LG.inneighbors(g::CycleGraph, v::Integer) = LG.outneighbors(g, v)
 
 # ---- neighbors iterator -----------------------------------
 
+function Base.size(nbs::OutNeighborVector{V, G}) where {V, G <: CycleGraph}
 
-function LG.iterate(iter::OutNeighborsIter{V, <:CycleGraph}) where {V}
-
-    g = iter.graph
+    g = nbs.graph
     T = eltype(g)
-    v = iter.vertex
-    nvg = nv(g)
+    nvg::T = nv(g)
 
-    nvg <= one(T) && return nothing
-    nvg == T(2)   && return (T(3) - v), zero(T)
-    v == one(T)   && return v + one(T),       nvg
-    v == nvg      && return one(T),     (v - one(T))
-
-    return (v - one(T)), (v + one(T))
+    nvg <= one(T) && return (0,)
+    nvg == T(2) && return (1,)
+    return (2,)
 end
 
-function LG.iterate(iter::OutNeighborsIter{V, <:CycleGraph}, state) where {V}
-
-    g = iter.graph
-    T = eltype(g)
-
-    state == zero(T) && return nothing
-    return state, zero(T)
-end
+# TODO propagate inbounds
+function Base.getindex(nbs::OutNeighborVector{V, G}, i::Int) where {V, G <: CycleGraph}
 
-function Base.length(iter::OutNeighborsIter{V, <:CycleGraph}) where {V}
+    i ∈ Base.OneTo(length(nbs)) || throw(BoundsError(nbs, i))
 
-    g = iter.graph
+    g = nbs.graph
     T = eltype(g)
-    nvg = nv(g)
-
-    nvg <= one(T) && return 0
-    nvg <= T(2) && return 1
+    v::T = nbs.vertex
+    nvg::T = nv(g)
 
-    return 2
+    if i == 1
+        v == one(T) && return T(2)
+        v == nvg && return one(T)
+        return v - one(T)
+    end
+    # i == 2
+    v == one(T) && return nvg
+    v == nvg && return nvg - T(1)
+    return v + one(T)
 end
 
+Base.IndexStyle(::Type{<:OutNeighborVector{V, G}}) where {V, G <: CycleGraph} = IndexLinear()
```

Comment:
is this required? Is `IndexLinear` not the default?

Reply:
No, it is `IndexCartesian`. Maybe I should put `IndexLinear` as default for these vectors though.